### PR TITLE
Update ZAP Dockerfile

### PR DESCRIPTION
**Pull request by InfoSec812**
#### What is this PR About?
This PR updates the containers-quickstarts for a reliable ZAP implementation. The Jenkins Agent Dockerfile.rhel7 works well even when using Multi-container Jenkins Pods. The Dockefile.rhel7-daemon file allows you to deploy ZAP as a persistent service which can then be used as needed from any pipeline.

#### How do we test this?
Check the `README.md` and the `Multi-Container_Pipeline_In_Jenkins_On_OpenShift.md` in the `jenkins-slaves/jenkins-slave-zap` directory.

cc: @redhat-cop/day-in-the-life

### CODE REVIEW

File: jenkins-slaves/jenkins-slave-zap/.openshift/templates/s2i-dockerfile-build.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        ref: ${SOURCE_REPOSITORY_REF}
+        uri: ${SOURCE_REPOSITORY_URL}
+      type: Git
+    strategy:
+      dockerStrategy:
+        dockerfilePath: Dockerfile.rhel7-daemon
+        from:
+          kind: DockerImage
+          name: ${BUILDER_IMAGE_NAME}
+      type: Source
```
Comment (haithamshahin333) [inline jenkins-slaves/jenkins-slave-zap/.openshift/templates/s2i-dockerfile-build.yml]: I believe this should be type: Docker

Comment (InfoSec812) [reply jenkins-slaves/jenkins-slave-zap/.openshift/templates/s2i-dockerfile-build.yml]: Probably correct, but in practice it doesn't seem to matter when I tested it on S11. I will try it with `type: Docker` and see if there is a differnce.

**Comment by etsauer** [comment]
@InfoSec812 will you rebase your branch so that all of the automated testing we've added can be used here?

**Comment by InfoSec812** [comment]
@etsauer Not sure how to accomplish that. Is there a reference?

**Comment by InfoSec812** [comment]
@etsauer Sorry, not quite awake. Upstream changes have been merged.

File: jenkins-slaves/jenkins-slave-zap/.openshift/templates/s2i-dockerfile-build.yml (hunk 1 above)
Comment (haithamshahin333) [reply jenkins-slaves/jenkins-slave-zap/.openshift/templates/s2i-dockerfile-build.yml]: Reviewing this again as I may be utilizing this image soon - is there a reason we are not utilizing the [generic jenkins build template](https://github.com/redhat-cop/containers-quickstarts/blob/master/.openshift/templates/jenkins-slave-generic-template.yml)

**Comment by haithamshahin333** [comment]
Currently building the Dockerfile.rhel7 agent image and seeing the following error:

`error: build error: API error (500): {"message":"invalid reference format"} openshift build`

**Comment by garethahealy** [comment]
/test

@InfoSec812 ; are you able to rebase/fix conflicts? if you don't the have time currently, can you sync up with @springdo since hes also got a PR open for the same issue as you've already noticed.
- https://github.com/redhat-cop/containers-quickstarts/pull/237

**Pull request closed without merging**